nx/lib/nx/random.ex

```diff
@@ -601,6 +601,242 @@ defmodule Nx.Random do
+          [3, 4, 5],
+          [6, 7, 8],
+          [3, 4, 5]
+        ]
+      >
+      iex> {result, _key} = Nx.Random.choice(k, t, samples: 4, axis: 0, replace: false) # without replacement
+      iex> result
+      #Nx.Tensor<
+        s64[4][3]
+        [
+          [3, 4, 5],
+          [9, 10, 11],
+          [6, 7, 8],
+          [0, 1, 2]
+        ]
+      >
+
+  If no axis is specified, the tensor is flattened:
+
+      iex> k = Nx.Random.key(2)
+      iex> t = Nx.iota({3, 2})
+      iex> {result, _key} = Nx.Random.choice(k, t, samples: 6) # with replacement
+      iex> result
+      #Nx.Tensor<
+        s64[6]
+        [5, 0, 0, 4, 0, 3]
+      >
+      iex> {result, _key} = Nx.Random.choice(k, t, samples: 6, replace: false) # without replacement
+      iex> result
+      #Nx.Tensor<
+        s64[6]
+        [2, 0, 4, 5, 1, 3]
+      >
+  """
+  deftransform choice(key, tensor), do: choice_uniform(key, tensor, [])
+  deftransform choice(key, tensor, opts) when is_list(opts), do: choice_uniform(key, tensor, opts)
+
+  @doc """
+  Generates random samples from a tensor with specified probabilities.
+
+  The probabilities tensor must have the same size as the axis along
+  which the samples are being taken. If no axis is given, the size
+  must be equal to the input tensor's size.
+
+  #{@choice_options}
+
+  ## Examples
+
+      iex> k = Nx.Random.key(1)
+      iex> t = Nx.iota({4, 3})
+      iex> p = Nx.tensor([0.1, 0.7, 0.2])
+      iex> {result, _key} = Nx.Random.choice(k, t, p, samples: 5, axis: 1) # with replacement
+      iex> result
+      #Nx.Tensor<
+        s64[4][5]
+        [
+          [1, 1, 1, 1, 0],
+          [4, 4, 4, 4, 3],
+          [7, 7, 7, 7, 6],
+          [10, 10, 10, 10, 9]
+        ]
+      >
+      iex> {result, _key} = Nx.Random.choice(k, t, p, samples: 3, axis: 1, replace: false) # without replacement
+      iex> result
+      #Nx.Tensor<
+        s64[4][3]
+        [
+          [1, 2, 0],
+          [4, 5, 3],
+          [7, 8, 6],
+          [10, 11, 9]
+        ]
+      >
+
+  If no axis is specified, the tensor is flattened.
+  Notice that in the first case we get a higher occurence
+  of the entries with bigger probabilities, while in the
+  second case, without replacements, we get those samples
+  first.
+
+      iex> k = Nx.Random.key(2)
+      iex> t = Nx.iota({2, 3})
+      iex> p = Nx.tensor([0.01, 0.1, 0.19, 0.6, 0.05, 0.05])
+      iex> {result, _key} = Nx.Random.choice(k, t, p, samples: 10) # with replacement
+      iex> result
+      #Nx.Tensor<
+        s64[10]
+        [2, 1, 3, 3, 3, 1, 3, 3, 1, 2]
+      >
+      iex> {result, _key} = Nx.Random.choice(k, t, p, samples: 6, replace: false) # without replacement
+      iex> result
+      #Nx.Tensor<
+        s64[6]
+        [3, 1, 2, 5, 4, 0]
+      >
+  """
+  deftransform choice(key, tensor, p), do: choice_probabilities(key, tensor, p)
+  deftransform choice(key, tensor, p, opts), do: choice_probabilities(key, tensor, p, opts)
```

Comment:
```suggestion
  deftransform choice(key, tensor, p, opts) when is_list(opts), do: choice_probabilities(key, tensor, p, opts)
```

Options seem required...